From the GCC 4.7 porting guide

  Many of the standard C++ library include files have been edited to no
  longer include <unistd.h> to remove namespace pollution.

  As such, C++ programs that used functions including truncate, sleep or
  pipe without first including <unistd.h> will no longer compile.

  Fixing this issue is easy: just include <unistd.h>.

--- a/src/sockwrap/sockwrap.cpp
+++ b/src/sockwrap/sockwrap.cpp
@@ -16,6 +16,8 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
+#include <unistd.h>
+
 #include "sockwrap.h"
 #include "sockwrap_plain.h"
 #ifdef HAVE_TLS
